server/app/mutations/grid_services/common.rb

```diff
@@ -261,6 +261,7 @@ def common_validations
               integer :initial_delay, default: 10
             end
           end
+          string :stop_grace_period, matches: Duration::PATTERN
```

Comment:
Should the pattern be anchored with `/\A...\z/`? If this is supposed to support things like `1h30m`, then the pattern would also need to repeat with `/\A(...)+\z/`. This will now accept a string that also contains other gibberish, which will get ignored.

If the user enters something like `5 m 30s`, then it will accept that input, but use a 30s timeout. Configuration typos like that should lead to errors, not unexpected behavior.